engine/src/sol/retry_rpc.rs

```diff
@@ -208,9 +209,34 @@ impl SolRetryRpcApi for SolRetryRpcClient {
 				Box::pin(move |client| {
 					let encoded_transaction = encoded_transaction.clone();
 					#[allow(clippy::redundant_async_block)]
-					Box::pin(
-						async move { client.send_transaction(encoded_transaction, config).await },
-					)
+					Box::pin(async move {
```

Comment:
you could write a test for this by mocking the client to return error or Ok responses at particular times

Reply:
after a chat we decided that it's not worth it as it needs a bunch of logic for mocks for little benefit.